### PR TITLE
make okhttp client wrapper into a non implicit class

### PR DESCRIPTION
**Pull request by johnduffell**
This PR makes the existing implicit OKhttpclient wrapper into a normal class and `Request => M[Response]` into a normal trait with a method on it.

This is a precursor to passing the identity id (`LogPrefix`) in in the next PR, so that we can log http messages against the appropariate identity id so they are searchable.

Tested in CODE and OK /me and /me/mma endpoints

### CODE REVIEW

File: membership-attribute-service/app/services/stripe/BasicStripeService.scala [hunk 1]
```diff
@@ -15,7 +15,5 @@ trait BasicStripeService {
 
   def fetchPaymentMethod(customerId: String): Future[CustomersPaymentMethods]
 
-  def fetchEvent(id: String): Future[Stripe.Event[StripeObject]]
```
Comment (johnduffell) [inline membership-attribute-service/app/services/stripe/BasicStripeService.scala:18]: unused, missed in previous prs

File: membership-attribute-service/app/services/subscription/SubscriptionService.scala [hunk 2]
```diff
@@ -39,8 +39,6 @@ trait SubscriptionService {
 
   def jsonSubscriptionsFromContact(contact: ContactId)(implicit logPrefix: LogPrefix): Future[Disjunction[String, List[JsValue]]]
 
-  def jsonSubscriptionsFromAccount(accountId: AccountId): Future[Disjunction[String, List[JsValue]]]
```
Comment (johnduffell) [inline membership-attribute-service/app/services/subscription/SubscriptionService.scala:42]: unused, missed in previous prs

File: membership-attribute-service/app/services/zuora/rest/SimpleClient.scala [hunk 3]
```diff
@@ -60,20 +61,12 @@ case class SimpleClient(
   def jsonBody(in: JsValue): RequestBody = RequestBody.create(MediaType.parse("application/json"), in.toString())
 
   def get[B](url: String)(implicit r: Reads[B]): Future[String \/ B] =
-    run(authenticated(url).get.build).map(parseResponse(_)(r))
-
-  def getJson(url: String): Future[String \/ JsValue] =
```
Comment (johnduffell) [inline membership-attribute-service/app/services/zuora/rest/SimpleClient.scala:65]: some unused methods missed from previous prs

File: membership-common/src/main/scala/com/gu/okhttp/RequestRunners.scala [hunk 4]
```diff
@@ -1,55 +1,58 @@
...
-  def configurableFutureRunner(timeout: Duration)(implicit ec: ExecutionContext): Request => Future[OkResponse] = {
+   * https://github.com/square/okhttp/wiki/Recipes#timeouts
+   */
+  def configurableFutureRunner(timeout: Duration)(implicit ec: ExecutionContext): FutureHttpClient = {
     val seconds: Int = timeout.toSeconds.toInt
-    client.newBuilder().readTimeout(seconds, TimeUnit.SECONDS).build().execute
+    new FutureHttpClient(client.newBuilder().readTimeout(seconds, TimeUnit.SECONDS).build())
   }
 
-  /** Logging runner with the default read timeout (10 seconds).
```
Comment (johnduffell) [inline membership-common/src/main/scala/com/gu/okhttp/RequestRunners.scala:34]: logging stuff was not used any more, and I didn't want to get my head around ReaderT at this stage

File: membership-common/src/main/scala/com/gu/salesforce/Scalaforce.scala [hunk 5]
```diff
@@ -15,33 +15,10 @@ import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
-case class SFResponse(Success: Boolean, ErrorString: Option[String])
```
Comment (johnduffell) [inline membership-common/src/main/scala/com/gu/salesforce/Scalaforce.scala:18]: unused stuff missed in prevous prs

File: membership-common/src/test/scala/com/gu/memsub/subsv2/services/CatalogServiceTest.scala [hunk 6]
```diff
@@ -33,7 +34,7 @@ class CatalogServiceTest extends Specification {
 object CatalogServiceTest {
 
   def client(path: String) = {
-    val runner = (r: Request) =>
+    val runner: HttpClient[Id] = (r: Request) =>
```
Comment (johnduffell) [inline membership-common/src/test/scala/com/gu/memsub/subsv2/services/CatalogServiceTest.scala:37]: giving it a type here explicitly makes scala convert the function to the trait automatically (single abstract method)

**Review by rupertbates — APPROVED** [review]
👍

**Pull request merged**